### ISSUE TITLE
26002 Segmentation fault

### ISSUE THREAD

**Issue by jiachangliu** [horacepan/MLGkernel#2]
Hi,

I can compile with no problems. However, when I actually run the go.sh file, there comes a segmentation fault. Do you have any idea where the error might be?

`++ pwd
+ BASE=/xxx/MLGkernel
+ dset=MUTAG
+ data=/xxx/MLGkernel/../data/MUTAG.txt
+ feats=/xxx/MLGkernel/..//data/MUTAG_nodelabels.txt
+ save=/xxx/MLGkernel//data/results/output.txt
+ mkdir -p /xxx/MLGkernel/data/results/
+ python preprocess.py MUTAG
+ for r in 1 2 3 4
+ for l in 1 2 3 4
+ for g in 0.01 0.1 1
+ for e in 0.01 0.1 1
+ cd MLGkernel
+ ./runMLG -d /xxx/MLGkernel/../data/MUTAG.txt -f /xxx/MLGkernel/..//data/MUTAG_nodelabels.txt -s /xxx/MLGkernel//data/results/output.txt -r 1 -l 1 -e 0.01 -g 0.01 -t 32 -m 1
Current parameter settings:
    -eta            : 0.01
    -gamma          : 0.01
    -radius         : 1
    -levels         : 1
    -grow_or_double : grow
    -data_path      : /xxx/MLGkernel/../data/MUTAG.txt
    -features_path  : /xxx/MLGkernel/..//data/MUTAG_nodelabels.txt
    -save_path      : /xxx/MLGkernel//data/results/output.txt
    -num_threads    : 32
Done loading dataset
Computing discrete features
num features: 7
./go.sh: line 27: 26002 Segmentation fault      ./runMLG -d $data -f $feats -s $save -r $r -l $l -e $e -g $g -t 32 -m 1`

**Comment by horacepan** [comment]
This is very odd. I ran the bash script with current parameters printed and I can't reproduce the seg fault. That the parameter settings printed means the code at least started computing the kernel. Can you figure out the exact line that the segfault happens with gdb (or by adding print statements everywhere)?

**Comment by jiachangliu** [comment]
Hello thank you very much for your quick reply. I have fixed the problem in my code. It's nothing to do with the C++ code but the python preprocessing code I wrote myself.

It turns out the networkx library I'm using has been updated. The syntax "G.node[]" has been updated to "G.nodes[]". This led to the incorrect format of the input I gave to your program. This leads in turn to iterating out of bounds for the loop in your C++ program. Now everything is working. Thanks again for the quick reply.
